pandas/io/parsers.py

```diff
@@ -1998,11 +2032,20 @@ def __init__(self, src, **kwds):
 
         self._implicit_index = self._reader.leading_cols > 0
 
-    def close(self):
+    def close(self) -> None:
+        if self._original_handle or isinstance(self.source, str):
```

Comment:
can you comment on the cases as appropriate

Reply:
done